src/bin/cargo/cli.rs

```diff
@@ -473,6 +480,14 @@ See 'cargo help <command>' for more information on a specific command.\n",
                 .value_name("WHEN")
                 .global(true),
         )
+        .arg(
+            opt("directory", "Change to DIRECTORY before doing anything")
+                .short('C')
+                .value_name("DIRECTORY")
+                .value_hint(clap::ValueHint::DirPath)
+                .value_parser(clap::builder::ValueParser::path_buf())
+                .global(true),
```

Comment:
Is it safe to be `global`?  In clap that means it will propagate to subcommands but the processing of this is fairly specific.

Reply:
I think that was a misunderstanding on my part, in part resulting from me implementing this part before I did the rest and understood how the search directories are recorded.